base/linalg/diagonal.jl

```diff
@@ -106,6 +106,22 @@ end
 /{T<:Number}(D::Diagonal, x::T) = Diagonal(D.diag / x)
 *(Da::Diagonal, Db::Diagonal) = Diagonal(Da.diag .* Db.diag)
 *(D::Diagonal, V::AbstractVector) = D.diag .* V
+# To avoid ambiguity in the definitions below
+for uplo in (:LowerTriangular, :UpperTriangular)
+    @eval begin
+        (*)(A::$uplo, D::Diagonal) = $uplo(A.data * D)
+
+        function (*)(A::$(Symbol(:Unit, uplo)), D::Diagonal)
+            B = A.data * D
+            for i = 1:size(A, 1)
+                B[i,i] = D.diag[i]
+            end
+            return $uplo(B)
+        end
+    end
+end
+(*)(A::AbstractTriangular, D::Diagonal) =
+    error("this method should never get called. Please make a bug report.")
```

Comment:
This was necessary to avoid an ambiguity warning but with https://github.com/JuliaLang/julia/pull/16125 it's not necessary anymore. I'm working on an updated version.